tests/test_graphical_units/test_transform_matching_parts.py

```diff
@@ -25,3 +25,40 @@ def test_TransformMatchingDisplaysCorrect(scene):
     scene.wait(0.5)
     # Shift to make sure object isn't duplicated if moved
     scene.play(circle.animate.shift(DOWN))
+
+
+@frames_comparison(last_frame=False)
+def test_TransformMatchingTex(scene):
+    start = MathTex("A", "+", "B", "=", "C")
+    end = MathTex("C", "=", "B", "-", "A")
+
+    scene.add(start)
+    scene.play(TransformMatchingTex(start, end))
+
+
+@frames_comparison(last_frame=False)
+def test_TransformMatchingTex_FadeTransformMismatches(scene):
+    start = MathTex("A", "+", "B", "=", "C")
+    end = MathTex("C", "=", "B", "-", "A")
+
+    scene.add(start)
+    scene.play(TransformMatchingTex(start, end, fade_transform_mismatches=True))
+
+
+@frames_comparison(last_frame=False)
+def test_TransformMatchingTex_TransformMismatches(scene):
+    start = MathTex("A", "+", "B", "=", "C")
+    end = MathTex("C", "=", "B", "-", "A")
+
+    scene.add(start)
+    scene.play(TransformMatchingTex(start, end, transform_mismatches=True))
+
+
+@frames_comparison(last_frame=False)
+def test_TransformMatchingTex_FadeTransformMismatches_NothingToFade(scene):
+    # https://github.com/ManimCommunity/manim/issues/2845
+    start = MathTex("A", "->", "B")
+    end = MathTex("B", "->", "A")
```

Comment:
I'd very much like to replace `->` with actual `Tex` arrows:

```suggestion
    start = MathTex("A", r"\to", "B")
    end = MathTex("B", r"\to", "A")
```